atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/PersistentLockService.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.persistentlock;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+
+/**
+ * Provides endpoints for acquiring and releasing the Deletion Lock. This is intended to be used by backups and sweep,
+ * to ensure that only one of these operations is deleting data at once.
+ */
+@Path("/persistent-lock")
```

Comment:
As of now `LockStore.ROW_NAME` is a final variable `DeletionLock`. Do we want to design this in a way that allows for the LockStore to be used for more than just the deletion lock? If so, I think this API is wrong.

I would rather the acquire endpoint look more like
```
@POST
@Path("acquire/{lock}")
public LockEntry acquireLock(@PathParam("lock") String lock, @QueryParam String reason)
```
Where `lock` represents the actual unique lock purpose / row key. We could track "supported" locks, like `DeletionLock` and throw otherwise. I always saw `reason` as being more for the user than for the actual lock and may not even have to be required.

The release would then look like
```
@Post
@Path("acquire/{lock}")
public boolean releaseLock(@PathParam("lock") String lock)
```
and wouldn't even require the reason (we would just do the logic for it in the background). Though I could be convinced taking in the whole LockEntry is okay, if we were to give the response a special value on failure that told you what the reason is (if you supplied the right ID but wrong reason)

Thoughts @gsheasby / @jboreiko ?